sklearn/svm/base.py

```diff
@@ -168,8 +168,32 @@ def fit(self, X, y, sample_weight=None):
                              "boolean masks (use `indices=True` in CV)."
                              % (sample_weight.shape, X.shape))
 
-        if self.gamma == 'auto':
-            self._gamma = 1.0 / X.shape[1]
+        if self.gamma in ('scale', 'auto', 'auto_deprecated'):
+            if isinstance(X, sp.csr_matrix):
+                # std = sqrt(E[X^2] - E[X]^2)
+                X_std = np.sqrt((X.multiply(X)).mean() - (X.mean())**2)
+            else:
+                X_std = X.std()
+            if self.gamma == 'scale':
+                if X_std != 0:
+                    self._gamma = 1.0 / (X.shape[1] * X_std)
+                else:
+                    self._gamma = 1.0
+            else:
+                if (self.gamma == 'auto_deprecated' and
+                        self.kernel not in ('linear', 'precomputed') and
+                        not np.isclose(X_std, 1.0) and
+                        not callable(self.kernel)):
+                    # NOTE: when deprecation ends we need to remove explicitly
+                    # setting `gamma` in examples (also in tests). See
+                    # https://github.com/scikit-learn/scikit-learn/pull/10331
+                    # for the examples/tests that need to be reverted.
+                    warnings.warn("The default value of gamma will change "
+                                  "from 'auto' to 'scale' in version 0.22 to "
+                                  "account better for unscaled features. Set "
+                                  "gamma explicitly to 'auto' or 'scale' to "
+                                  "avoid this warning.", DeprecationWarning)
```

Comment:
Maybe a FutureWarning rather than a DeprecationWarning? We are not really deprecating anything, but warning about changing the default in the future.

Reply:
I suggested that here https://github.com/scikit-learn/scikit-learn/pull/10331#discussion_r163312773 but I don't remember why then I thought otherwise.

@jnothman opinion?